src/components/block/index.ts

```diff
@@ -257,7 +257,11 @@ export default class Block extends EventsDispatcher<BlockEvents> {
     super();
 
     this.name = tool.name;
-    this.id = id;
+    if (id) {
+      this.id = id;
+    } else {
+      this.id = _.generateBlockId();
+    }
```

Comment:
Looks like it is the same behaviour that we already have. Because `_.generateBlockId()` will be called only if the id won't be passed.